File: examples/node/02-index-and-search.mjs
// Build a knowledge-graph index and search it from Node.
//
//   npm install @hyper-light/vorpal-node
import {
  indexBuildAsync,
  indexBuildReportAsync,
  indexSearchAsync,
  indexSearchRanked,
} from '@hyper-light/vorpal-node'

const SRC = '.'
const IDX = '.vorpal/index'

// Incremental build — instant when nothing changed — off the event loop (every
// blocking call has an Async twin; the sync forms exist for scripts).
console.log(await indexBuildAsync(SRC, IDX))

// Structured numbers when you need them.
const report = await indexBuildReportAsync(SRC, IDX)
console.log(`indexed=${report.indexed} replayed=${report.skipped} errorFiles=${report.errorFiles}`)

// Hybrid search (rendered text; pass explain=true for per-hit provenance).
console.log(await indexSearchAsync(IDX, 'parse configuration', 5))

// One search, two orderings (requires an encoder: `vorpal enable semantic-f16`).
console.log(indexSearchRanked(IDX, 'retry backoff', 5))

File: examples/python/02_index_and_search.py
"""Build a knowledge-graph index and search it — the repo-scale half of vorpal_py.

    pip install vorpal-py
"""
import json
import vorpal_py

SRC = "."                       # any source tree
IDX = ".vorpal/index"           # default CLI-compatible location

# Build (incremental: instant when nothing changed). Returns the one-line report.
print(vorpal_py.index_build(SRC, IDX))

# Structured build report instead, when you want the numbers.
report = vorpal_py.index_build_report(SRC, IDX)
print(f"indexed={report.indexed} replayed={report.skipped} error_files={report.error_files}")

# Hybrid search (rendered text; explain=True appends per-hit provenance).
print(vorpal_py.index_search(IDX, "parse configuration", k=5))

# One search, two orderings (needs an encoder: vorpal enable semantic-f16).
ranked = vorpal_py.index_search_ranked(IDX, "retry backoff", k=5)
print(json.dumps(ranked, indent=2, default=str) if not isinstance(ranked, str) else ranked)
